pandas/core/accessor.py

```diff
@@ -291,8 +291,65 @@ def register_series_accessor(name):
     return _register_accessor(name, Series)
 
 
-@doc(_register_accessor, klass="Index")
 def register_index_accessor(name):
+    """
+    Register a custom accessor with an Index.
+
+    Parameters
+    ----------
+    name : str
+        Name used when calling the accessor after its registered.
+
+    Returns
+    -------
+    callable
+        A class decorator.
+
+    See Also
+    --------
+    register_dataframe_accessor: Register a custom accessor on DataFrame objects.
+    register_series_accessor: Register a custom accessor on Series objects.
+
+    Notes
+    -----
+    When accessed, your accessor will be initialized with the pandas_obj object
+    the user is interacting with. The code signature must be::
+
+        def __init__(self, pandas_obj):
+            # constructor logic
+        ...
+
+    >>> pd.Series(['a', 'b']).dt
```

Comment:
if you're making an Index-specific docstring, should you use an Index example here?